saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/Saml2PostAuthenticationRequest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.saml2.provider.service.authentication;
+
+import org.springframework.security.saml2.provider.service.registration.Saml2MessageBinding;
+
+import static org.springframework.security.saml2.provider.service.registration.Saml2MessageBinding.POST;
+
+/**
+ * Data holder for information required to send an {@code AuthNRequest} over a POST binding
+ * from the service provider to the identity provider
+ * https://www.oasis-open.org/committees/download.php/35711/sstc-saml-core-errata-2.0-wd-06-diff.pdf (line 2031)
+ *
+ * @see {@link Saml2AuthenticationRequestFactory}
+ * @since 5.3
+ */
+public class Saml2PostAuthenticationRequest extends AbstractSaml2AuthenticationRequest {
+
+	private Saml2PostAuthenticationRequest(
+			String issuer,
+			String samlRequest,
+			String relayState,
+			String destination) {
+		super(issuer, samlRequest, relayState, destination);
+	}
+
+	@Override
+	/**
+	 * @return {@link Saml2MessageBinding#POST}
+	 */
+	public Saml2MessageBinding getBinding() {
+		return POST;
+	}
+
+	public static Builder builder(Saml2AuthenticationRequestContext context) {
```

Comment:
Just leaving a note here to remember to rename this to follow Spring Security's naming conventions, e.g. `withFoo(Foo foo)`. We'd probably do `withAuthenticationRequestContext` here.

Reply:
Completed